pkg/config/validation/validation.go

```diff
@@ -1936,6 +1939,30 @@ func ValidateControlPlaneAuthPolicy(policy meshconfig.AuthenticationPolicy) erro
 	return fmt.Errorf("unrecognized control plane auth policy %q", policy)
 }
 
+func validatePolicyTargetReference(targetRef *type_beta.PolicyTargetReference) (v Validation) {
+	if targetRef == nil {
+		return
+	}
+	if targetRef.Name == "" {
+		v = appendErrorf(v, "policyTargetReference.name must be set")
+	}
+	// TODO: we should also validate that the namespace is not a root namespace
```

Comment:
I think the policyTargetReference namespace can be in the root namespace.

Reply:
Good catch. Just noticed this development w.r.t that topic: https://github.com/istio/api/pull/2888#issuecomment-1682822033

> even though this is a bad idea, we shouldn't stop users from doing it with a one-off API exception. However, if the user creates a targetRef policy in the root namespace, we should not apply it mesh-wide as we do for other Istio resources

For validation purposes we don't care. I'll update.

Reply:
I think @howardjohn convinced me of this too. Just because it's not a good idea to have, for example, a waypoint in the root namespace, it's valid from an API standpoint